update/Pos/Update/Logic/Local.hs

```diff
@@ -251,11 +256,10 @@ processVote vote = processSkeleton $ UpdatePayload Nothing [vote]
 -- tries to leave as much data as possible. It assumes that
 -- 'blkSemaphore' is taken.
 usNormalize :: (USLocalLogicMode ctx m) => m ()
-usNormalize =
-    withUSLock $ do
-        tip <- DB.getTip
-        stateVar <- mvState <$> views (lensOf @UpdateContext) ucMemState
-        atomically . writeTVar stateVar =<< usNormalizeDo (Just tip) Nothing
+usNormalize = do
```

Comment:
Here's another case where the typing trick to enforce that we hold the lock would be useful, rather than just having it as a comment. It's easy to make mistakes with this or in later refactoring.